react/ProductGifts.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { FC } from 'react'
+import {
+  useResponsiveValue,
+  MaybeResponsiveInput,
+} from 'vtex.responsive-values'
+
+import { ProductGiftsContextProvider } from './ProductGiftsContext'
+
+interface Props {
+  maxVisibleItems: MaybeResponsiveInput<number | 'showAll'>
+}
+
+const ProductGifts: FC<Props> = ({ children, maxVisibleItems = 'showAll' }) => {
+  const staticMaxVisibleItems = useResponsiveValue<number | 'showAll'>(
+    maxVisibleItems
+  )
+
+  return (
+    <ProductGiftsContextProvider maxVisibleItems={staticMaxVisibleItems}>
+      {children}
```

Comment:
I think it's useful to have a wrapper div. What do you think? Or flex-layout fulfill this?

Reply:
You could wrap this with a `flex-layout`, but it might be a good idea to have a handle, in case the user wants to use this inside a `flex-layout` with another block and not have it wrapped again just for CSS reasons.